### DISCUSSION TITLE
Module '"hono"' declares 'Hono' locally, but it is not exported.

### DISCUSSION THREAD

**Discussion by althafdaa** [honojs/hono#3668] [Q&A]
Hi im trying to setup hono with remix, and i keep getting typescript error 
`Module '"hono"' declares 'Hono' locally, but it is not exported.`

look like it caused by `compilerOptions.moduleResolutions` but if i changed it my remix import just break (import with tsconfigpaths broken or some modules not found)
am still not pretty sure what happened

## Step to Reproduce
1. install default remix app
```
pnpm create remix@latest hono-remix
```
2. add hono
```
cd hono-remix
pnpm add hono@latest
```
3. Import hono
```
import {Hono} from 'hono'
```
this is how tsconfig.ts,
just default remix
```
{
    "compilerOptions": {
        "lib": [
            "dom",
            "dom.iterable",
            "es2022"
        ],
        "types": [
            "@remix-run/node",
            "vite/client"
        ],
        "isolatedModules": true,
        "esModuleInterop": true,
        "jsx": "react-jsx",
        "module": "node16",
        "moduleResolution": "node16",
        "resolveJsonModule": true,
        "target": "es2022",
        "strict": true,
        "allowJs": true,
        "skipLibCheck": true,
        "forceConsistentCasingInFileNames": true,
        "baseUrl": "./",
        "paths": {
            "~/*": [
                "./app/*"
            ]
        },
        "noEmit": true
    },
    "files": [
        "./tailwind.config.ts",
        "./vite.config.ts",
        "./hono/index.ts",
        "./app/entry.client.tsx",
        "./app/entry.server.tsx",
        "./app/root.tsx",
        "./app/routes/_index.tsx"
    ],
    "include": [
        "**/*.ts",
        "**/*.tsx",
        "**/.server/**/*.ts",
        "**/.server/**/*.tsx",
        "**/.client/**/*.ts",
        "**/.client/**/*.tsx"
    ]
}
```

<img width="772" alt="image" src="https://github.com/user-attachments/assets/94f76119-ad70-47c6-b010-08ad242fc99c">

## Depedencies
- hono:v4.6.10
- @remix-run/node:v2.14.0
- vite:v5.10
- typescript:v5.1.6

Anyone had simillar issue? thanks

**Comment by marcomuser** [comment]
Do you have type: module in all your package.json files? I don't think hono works in combination with commonjs and module: node16. So just to exclude this issue double check that your project is esm based.

**Comment by joodaloop** [comment]
Having this issue, fixed by putting my server folder inside of Remix's /app directory as a hidden folder (name starts with '.')
